src/components/SearchBar/AbstractSearchBar.js

```diff
@@ -0,0 +1,149 @@
+import  React from 'react';
+import "react-filter-box/lib/react-filter-box.css";
+import "./index.css";
+
+import ReactFilterBox, { SimpleResultProcessing, GridDataAutoCompleteHandler} from "react-filter-box";
+
+ import CheckCircle  from 'react-icons/lib/fa/check-circle';
+ import TimeCircle  from 'react-icons/lib/fa/times-circle';
+ import SearchIcon  from 'react-icons/lib/fa/search';
+ import columnAccessor from "../../utils/columnAccessor";
+ 
+
+//extend this class to add your custom operator
+export class CustomAutoComplete extends GridDataAutoCompleteHandler {
```

Comment:
I think react is more particular in using composition rather then inheritance. See [here](https://facebook.github.io/react/docs/composition-vs-inheritance.html) Is it possible to change it as such?  I happen to agree!

Reply:
@natabal Let me check it and see the possibility to convert it into composition instead of Inheritence.